### PR TITLE
docs(app-menu): refactor storybook docs

### PR DESCRIPTION
**Pull request by ianaa**
First of all: sorry for the size of the diff. I honestly just went alphabetically with updating the docs, and this became sort of a monster. 
The big change here is that the ArgTypes component from storyybook DocBlocks not longer supports subcomponents (or table tabs) to show props and slots of multiple components on one docs page.
My solution here is to split one big mdx file into multiple ts ones and rely on autogenerated docs.
The stories are mostly the same, but I tried to reuse the templates a bit.

Apart from the docs changes, I added a `controls` prop to `a-app-multi-button` (this was overlooked before).

### CODE REVIEW

**Comment by manuatarchilogic** [comment]
the subpages of AppMenuItem don't work? https://649d652b7201581ed9f63642-sxjgmmbsft.chromatic.com/?path=/docs/components-appmenu-appmenuitem--docs

Same for https://649d652b7201581ed9f63642-sxjgmmbsft.chromatic.com/?path=/docs/components-appmenu-appmenumultibutton--docs

**Comment by ianaa** [comment]
> the subpages of AppMenuItem don't work? https://649d652b7201581ed9f63642-sxjgmmbsft.chromatic.com/?path=/docs/components-appmenu-appmenuitem--docs
> 
> Same for https://649d652b7201581ed9f63642-sxjgmmbsft.chromatic.com/?path=/docs/components-appmenu-appmenumultibutton--docs

@manuatarchilogic not sure what to say, loads for me both for the deployed links you provided and locally. also in safari!

**Comment by luk-zap** [comment]
Screenshot 2023-11-27 at 13 42 37

Is it the intended behavior? Based on the description I would expect the button should stay blueish

**Comment by luk-zap** [comment]
I noticed some issues, probably out of the scope of this PR, but I think worth mention
1. Cannot scroll to the first item

https://github.com/archilogic-com/honeycomb/assets/118187981/943a3c01-fa82-40bc-b27a-b40f69c85251

2. Repositioning doesn't work properly?

Screenshot 2023-11-27 at 13 49 16

3.  The arrow jumps when I hover over the `AppMenuMultiButton`
4. I can open the menu by clicking on the arrow only

https://github.com/archilogic-com/honeycomb/assets/118187981/20f3f6ea-84e0-4c4d-8f6f-4c083824a5a8

5. `AppMenuMultiButton` I don't see any differences between `Primary`, `Pressed` and `Menu open` stories

**Comment by manuatarchilogic** [comment]
> > the subpages of AppMenuItem don't work? https://649d652b7201581ed9f63642-sxjgmmbsft.chromatic.com/?path=/docs/components-appmenu-appmenuitem--docs
> > Same for https://649d652b7201581ed9f63642-sxjgmmbsft.chromatic.com/?path=/docs/components-appmenu-appmenumultibutton--docs
> 
> @manuatarchilogic not sure what to say, loads for me both for the deployed links you provided and locally. also in safari!

it does now too for me 🤷

**Comment by ianaa** [comment]
> I noticed some issues, probably out of the scope of this PR, but I think worth mention
> 
@luk-zap thanks for being so thorough! in short, all of these seem to be non-issues.
- _pressed state off_: if `pressed` is specifically set to false on the menu button, the button won't be highlighted in blue even when the menu is open
- _scroll and repositioning_: these don't work reliably on the `Docs` page, but should be ok on each story page (what was called 'canvas' in the previous storybook version)
- _app-menu-multi-button_:  it's a special button with two sections: the main one activates the "pressed" button state (like a toggle button) and the area behind the arrow opens the menu. it's used in editor's toolbar (e.g. for changing the element being drawn). a bit complicated but an established pattern in application menus (see figma's menus for example). The arrow moves on hover on purpose.  You don't see the difference between Primary, Pressed and Menu open storis on the docs page, but you should see it on each story's canvas page. The reason for this is that I use a `play` function for some interaction testing and it's not being triggered in the docs page (by design).

**Review by luk-zap — APPROVED** [review]

File: src/stories/AppMenu/AppMenuButton.stories.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import type { Meta, StoryObj } from '@storybook/vue3'
+import type { ComponentProps } from 'vue-component-type-helpers'
+import { AAppMenu, AAppMenuItem, AAppMenuButton, AButton, AIcon } from '../../components'
+import { menuStoriesDecorators, menuStoriesParameters, openMenu } from './shared'
+
+/**
+ * `<a-app-menu-button>` (aka `<a-popup-button>`) renders an AButton component by default,
+ * and accepts[ AButton's props](path=/docs/components-button--docs) in addition to APopupButton props.
```
Comment (romiekos) [inline src/stories/AppMenu/AppMenuButton.stories.ts]:
```suggestion
 * and accepts [AButton's props](path=/docs/components-button--docs) in addition to APopupButton props.
```

**Review by romiekos — APPROVED** [review]
Nice refactoring!

**Pull request merged**